pandas/core/indexes/interval.py

```diff
@@ -785,7 +788,10 @@ def get_loc(self, key, method=None):
                 left, right = _get_interval_closed_bounds(key)
                 return self._engine.get_loc_interval(left, right)
             else:
-                return self._engine.get_loc(key)
+                try:
+                    return self._engine.get_loc(key)
+                except TypeError:
+                    raise KeyError('No engine for key {!r}'.format(key))
```

Comment:
No need to mention "engine" here (that is something internal to pandas, while this error message will be visible for users)